### ISSUE TITLE
Numeric instability in MPPI

### ISSUE THREAD

**Issue by brian-bfz** [WangYixuan12/pytorch_mpc#1]
Softmax returns NaN when reward_seqs * self.reward_weight is too large. Need reward_seqs = reward_seqs - torch.max(reward_seqs) in optimize_action_mppi
